allennlp/training/trainer_pieces.py

```diff
@@ -78,7 +80,9 @@ def from_params(
         model.extend_embedder_vocab()
 
         # Initializing the model can have side effect of expanding the vocabulary
-        vocab.save_to_files(os.path.join(serialization_dir, "vocabulary"))
+        # Save the vocab only in the master
+        if not dist.is_initialized() or dist.get_rank() == 0:
+            vocab.save_to_files(os.path.join(serialization_dir, "vocabulary"))
```

Comment:
This needs to be done in case of non-distributed training as the vocab creation wouldn't be done up until this point. Whereas in the distributed case, it is done before spawning the workers since the entire view of the data is not available to the workers.

Reply:
Ah, sorry. I think I was confused as to the original purpose of "# Initializing the model can have side effect of expanding the vocabulary", but that makes sense now.